std/assembly/util/memory.ts

```diff
@@ -201,17 +201,16 @@ export function memset(dest: usize, c: u8, n: usize): void { // see: musl/src/st
   } else {
     // fill head and tail with minimal branching
     if (!n) return;
-    let dend = dest + n - 4;
     store<u8>(dest, c);
-    store<u8>(dend, c, 3);
+    store<u8>(dest + n - 1, c);
     if (n <= 2) return;
-    store<u8>(dest, c, 1);
-    store<u8>(dest, c, 2);
-    store<u8>(dend, c, 2);
-    store<u8>(dend, c, 1);
+    store<u8>(dest + 1, c);
+    store<u8>(dest + 2, c);
+    store<u8>(dest + n - 2, c);
+    store<u8>(dest + n - 3, c);
     if (n <= 6) return;
-    store<u8>(dest, c, 3);
-    store<u8>(dend, c);
+    store<u8>(dest + 3, c);
+    store<u8>(dest + n - 4, c);
```

Comment:
How about revert only addresses which contain subtractions? Like `dest + n - 4` and `dest + n - 12` but rest of changes keep the same. This `store<u8>(dest + 1, c)` could be safely reverted to `store<u8>(dest, c, 1)`

wdyt?

Reply:
According to `memory.fill` semantics that should be acceptable

> A trap occurs if: the destination offset plus size is greater than the length of the target memory

as long as we do not otherwise use `memset` like its libc counterpart I think, which I believe we don't do.